component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/SchemaImpl.java

```diff
@@ -85,11 +96,101 @@ public Builder toBuilder() {
                         .entrySet()
                         .stream()
                         .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
-        this.getAllEntries().forEach(builder::withEntry);
+        getEntriesOrdered().forEach(builder::withEntry);
         return builder;
     }
 
-    public static class BuilderImpl implements Schema.Builder {
+    @Override
+    @JsonbTransient
+    public List<Entry> getEntriesOrdered() {
+        return getAllEntries().sorted(entriesOrder).collect(Collectors.toList());
+    }
+
+    @Override
+    @JsonbTransient
+    public List<Entry> getEntriesOrdered(final Comparator<Entry> comparator) {
+        return getAllEntries().sorted(comparator).collect(Collectors.toList());
```

Comment:
This could be defined as default in interface ?

Reply:
done